packages/data-layer/src/data-layer.ts

```diff
@@ -69,8 +99,153 @@ export class DataLayer {
       collections?.googleSheetsUrl === undefined
         ? { type: "hardcoded" }
         : { type: "google-sheet", url: collections.googleSheetsUrl };
+    this.gsIndexerEndpoint = indexer.baseUrl;
+  }
+
+  /**
+   * Allo v1 & v2 builder queries
+   */
+
+  /**
+   * Gets a project by its ID.
+   *
+   * @example
+   * Here is an example:
+   * ```
+   * const project = await dataLayer.getProjectById({
+   *  projectId: "0x1234",
+   *  chainId: 1,
+   *  alloVersion: "allo-v2",
+   * });
+   * ```
+   * @param projectId - the ID of the project to return.
+   * @param chainId - the network ID of the chain.
+   * @param alloVersion - the version of Allo to use.
+   *
+   * @returns v2Project | Project
+   */
+  async getProjectById({
+    projectId,
+    chainId,
+    alloVersion,
+  }: {
+    projectId: string;
+    chainId: number;
+    alloVersion: AlloVersion;
+  }): Promise<{ project: v2Project } | null> {
+    const requestVariables = {
+      alloVersion,
+      projectId,
+      chainId,
+    };
+
+    const response: { projects: v2Project[] } = await request(
+      this.gsIndexerEndpoint,
+      getProjectById,
+      requestVariables,
+    );
+
+    if (response.projects.length === 0) return null;
+
+    const project = response.projects[0];
+
+    return { project };
   }
 
+  /**
+   * getProjects() returns a list of projects.
+   *
+   * @param chainIds
+   * @param first
+   * @param alloVersion
+   *
+   * @returns v2Projects[] | null
+   */
+  async getProjects({
+    chainIds,
+    first,
+    alloVersion,
+  }: {
+    chainIds: number[];
+    first: number;
+    alloVersion: AlloVersion;
+  }): Promise<{ projects: v2Project[] } | null> {
+    const projects: v2Project[] = [];
+
+    for (const chainId of chainIds) {
+      const requestVariables = {
+        alloVersion,
+        first,
+        chainId,
+      };
+
+      const profilesData: v2Project = await request(
+        this.gsIndexerEndpoint,
+        getProjects,
+        requestVariables,
+      );
+
+      projects.push(profilesData);
+    }
+
+    return {
+      projects,
+    };
+  }
+
+  // getProjectsByAddress
+  /**
+   * getProjectsByAddress() returns a list of projects by address.
+   * @param address
+   * @param chainId
+   * @param role
+   */
+  async getProjectsByAddress({
+    address,
+    alloVersion,
+    chainId,
+  }: {
+    address: string;
+    alloVersion: AlloVersion;
+    chainId: number;
+  }): Promise<ProjectEventsMap | undefined> {
```

Comment:
```suggestion
  }): Promise<ProjectEventsMap | null> {
```

the convention in the data layer is to return null to indicate empty value, it would be good to keep consistency, note that you might have to update the call site